backend/schedulerd/proxy_check.go

```diff
@@ -34,7 +35,7 @@ func matchEntities(entities []EntityCacheValue, proxyRequest *types.ProxyRequest
 
 	for i, result := range results {
 		if result {
-			matched = append(matched, entities[i].Entity)
+			matched = append(matched, entities[i].Resource.(*corev2.Entity))
```

Comment:
I suppose `len(results) == len(entities)` is checked above, but this definitely threw me off using `entities[i]` while looping through `results`. Not sure if there is a more fluid approach, just thought I'd mention it.

Reply:
To be honest I'm not sure how we could do that in a more simplistic way without changing the signature of `js.MatchEntities` so it returns the matched entities instead, which I'd like to avoid!